ext/net/net.rs

```diff
@@ -0,0 +1,79 @@
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use fqdn::FQDN;
+use std::net::Ipv4Addr;
+use std::net::Ipv6Addr;
+use std::str::FromStr;
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub enum Host {
+  FQDN(FQDN),
+  Ipv4(Ipv4Addr),
+  Ipv6(Ipv6Addr),
+}
+
+impl Host {
+  pub fn from_str(host: &str, origin_host: &str) -> Result<Self, AnyError> {
+    if let Ok(ipv6) = host.parse::<Ipv6Addr>() {
+      return Ok(Host::Ipv6(ipv6));
+    }
+
+    let host = FQDN::from_str(host)
+      .with_context(|| format!("Failed to parse host: {}\n", origin_host))?;
+    let host_string = host.to_string();
+
+    if let Ok(ipv4) = host_string.parse::<Ipv4Addr>() {
+      return Ok(Host::Ipv4(ipv4));
+    }
+
+    return Ok(Host::FQDN(host));
+  }
+
+  pub fn to_string(&self) -> String {
```

Comment:
You might have to implement `Display` trait instead, but I'm not sure about clippy